packages/block/src/block.ts

```diff
@@ -73,10 +73,29 @@ export class Block {
     }
 
     // parse transactions
+
+    // get common based on block's number
+    // since hardfork param may not be set
+    // (see TODO above)
+    let common = this._common
+    if (!common.hardfork()) {
+      const blockNumber = parseInt(this.header.number.toString('hex'), 16)
+      const chainId = this._common.chainId()
+      if (blockNumber) {
+        const hfs = this._common.activeHardforks(blockNumber)
+        const hf = hfs[hfs.length - 1].name
+        common = new Common(chainId, hf)
+      } else {
+        // fall back to default hf
+        common = new Common(chainId, 'petersburg')
```

Comment:
Maybe this should be a constant in `Common`

Reply:
`'petersburg'`, I mean

Reply:
yeah, maybe I can do in a follow-up PR since it's a default assumption across the broader monorepo like in the StateManager [here][1].

Can we export constants with common? It would be nice if we could import e.g. `DEFAULT_HARDFORK`. I know we also have have #702 that would also be great to get to.

[1]: https://github.com/ethereumjs/ethereumjs-vm/blob/master/packages/vm/lib/state/stateManager.ts#L46

Reply:
opened an issue for this https://github.com/ethereumjs/ethereumjs-vm/issues/822